### PR TITLE
New Added Movie app project

### PR DESCRIPTION
**Pull request by Nusrath-Jahan**

### CODE REVIEW

File: Javascript-2/Week-1/Movie App/arrayList.js [hunk 1]
```diff
@@ -0,0 +1,460 @@
+export const movies = [
```
Comment (vdaguenet) [inline Javascript-2/Week-1/Movie App/arrayList.js:1]: The `export` keyword throws an error:
image

If you want to use it, you must import the `arrayList.js` as a module in your HTML file

File: Javascript-2/Week-1/Movie App/index.html [hunk 2]
```diff
@@ -0,0 +1,200 @@
...
+      integrity="sha512-SnH5WK+bZxgPHs44uWIX+LLJAJ9/2PkPKZ5QiAj6Ta86w+fsb2TkcmfRyVX3pBnMFcV7oQPJkl9QevSCWr3W6A=="
+      crossorigin="anonymous"
+      referrerpolicy="no-referrer"
+    />
+    <script type="module" src="script.js" defer></script>
+    <script src="arrayList.js"></script>
+</head>
+<body>
+    <header>
+        <!-- <img
```
Comment (vdaguenet) [inline Javascript-2/Week-1/Movie App/index.html:20]: Please clean up your code of debugging comments before submitting your PR. It will make your files difficult to review.

File: Javascript-2/Week-1/Movie App/index.html [hunk 3]
```diff
@@ -0,0 +1,200 @@
...
+                </select></label
+              >
+              <p id="genres" ,class="filter">genres</p>
+            </fieldset>
+          </span> -->
+          <!-- <span> <i class="fa-solid fa-bell"></i></span> -->
+          <span>
+            <input
+              type="search"
+              ,
```
Comment (vdaguenet) [inline Javascript-2/Week-1/Movie App/index.html:62]: You don't need these commas between each attribute of an HTML element. I am commenting on it here, but I have seen them a bit everywhere.

File: Javascript-2/Week-1/Movie App/index.html [hunk 4]
```diff
@@ -0,0 +1,200 @@
...
+                  placeholder="Your password"
+                  ,
+                  pattern
+                  required /></label
+              ><br />
+              <button id="signIn">Sign In</button>
+              <p>If not registered! Please Sign up</p>
+              <button id="signUp">Sign Up</button><br />
+              <div id="signUpPage">
+                <label
```
Comment (vdaguenet) [inline Javascript-2/Week-1/Movie App/index.html:99]:
```suggestion
                <label>
                Enter Your First Name:
```
Please keep the closing `>` on the same line as the tag name.

File: Javascript-2/Week-1/Movie App/script.js [hunk 5]
```diff
@@ -0,0 +1,201 @@
...
+             <p><strong>Description:</strong> ${movie.description}</p>
+             <p><strong>Release Year:</strong> ${movie.movie_year}</p>
+             <p><strong>Director:</strong> ${movie.director}</p>
+             <p><strong>Actors:</strong> ${movie.actors.join(", ")}</p>
+         `;
+  movieContainer.appendChild(movieDetails);
+
+  // Adding the comment Section
+
+  function createCommentSection() {
```
Comment (vdaguenet) [inline Javascript-2/Week-1/Movie App/script.js:69]: Avoid declaring new functions inside another one. IMO `createCommentSection` can be defined outside of `createMovieCard`.

File: Javascript-2/Week-1/Movie App/script.js [hunk 6]
```diff
@@ -0,0 +1,201 @@
+import { movies } from "./arrayList.js";
+
+//////////////////////Streaming Function
+const nowStreaming = document.querySelector(".streaming");
+
+function createMovieCard(movie) {
```
Comment (vdaguenet) [inline Javascript-2/Week-1/Movie App/script.js:6]: This function is longer than 110 lines; it is way too long. You need to split it into smaller parts. This will simplify your work for debugging and allow your coworkers to easily understand your logic.

File: Javascript-2/Week-1/Movie App/script.js [hunk 7]
```diff
@@ -0,0 +1,201 @@
...
+
+function createMoviesPage() {
+  nowStreaming.innerHTML = "";
+  movies.forEach((movie) => {
+    const movieElement = createMovieCard(movie);
+    nowStreaming.appendChild(movieElement);
+  });
+}
+createMoviesPage();
+// JS 2
```
Comment (vdaguenet) [inline Javascript-2/Week-1/Movie App/script.js:129]: It's nice that you started working on the second part of the project. For the next time, please do it in a different PR. You can send your PR for the week N, checkout a new branch from it and then send a PR for the week N + 1.

```bash
git checkout main
git checkout -b javascript2-week1
# work on the assignment for the week 1
# submit the PR
git checkout -b javascript2-week2 # note that I am not starting this branch from main  but from javascript2-week1
# work on the assignment for the week 2
# submit the PR for week 2
```
Finally, when you create the PR for the week 2, choose javascript2-week1 as the base branch by click on the `base: main` button:
image

Comment (vdaguenet) [reply Javascript-2/Week-1/Movie App/script.js:129]: Or for simplicity, wait for the PR of week 1 to be merged before creating a new PR for week 2 (but still work on the week 2 assignment in a different branch).

File: Javascript-2/Week-1/Movie App/style.css [hunk 8]
```diff
@@ -0,0 +1,270 @@
...
+  display: flex;
+  justify-content: space-between;
+  align-content: center;
+  margin: 20px 0 20px 10px;
+  padding: 20px 0 0 50px;
+}
+section {
+    margin: 20px 0 20px 50px;
+  }
+  .site-name {
```
Comment (vdaguenet) [inline Javascript-2/Week-1/Movie App/style.css:16]:
```suggestion
.site-name {
```
Mind the indentation here and for the other classes below

**Review by vdaguenet — CHANGES_REQUESTED** [review]
The functionalities required for week 1 of the project work as expected. However there is still an error in the console that seems to come from the week 2 part:
image

Please spend a couple of minutes cleaning up your code before sending your assignment for review. And be careful about the indentation and the length of your functions.

File: Javascript-2/Week-1/Movie App/style.css (hunk 8 above)
Comment (Nusrath-Jahan) [reply Javascript-2/Week-1/Movie App/style.css]: Hi,
First of all, thank you very much for your valuable advice. I have made the corrections to my code as per your suggestions. I hadn't realized there were so many mistakes, and your comments were incredibly helpful. Thanks to your guidance, I was able to resolve the issues.
Best regards,
Nusrath

File: Javascript-2/Week-1/Movie App/index.html [hunk 9]
```diff
@@ -0,0 +1,71 @@
...
+                <br />
+
+              <button id="signIn">Sign In</button>
+
+              <p>If not registered! Please Sign up</p>
+
+              <button id="signUp">Sign Up</button><br />
+
+              <div id="signUpPage">
+<label>Enter Your First Name:<input type="text" id="firstName" required /></label>
```
Comment (vdaguenet) [inline Javascript-2/Week-1/Movie App/index.html:46]: Don't forget to indent here

File: Javascript-2/Week-1/Movie App/script.js [hunk 10]
```diff
@@ -0,0 +1,197 @@
...
+    if (!found) {
+      nowStreaming.innerHTML = `<p>Not Found</p>`;
+    }
+  });
+});
+
+const login = document.getElementById("loginButton");
+const loginPage = document.getElementById("logIn");
+
+login.addEventListener("mouseover", function onclick() {
```
Comment (vdaguenet) [inline Javascript-2/Week-1/Movie App/script.js:175]:
```suggestion
login.addEventListener("mouseover", function onMouseOver() {
```
It's confusing to have a function called `onclick` for a `mouseover` event.

File: Javascript-2/Week-1/Movie App/style.css [hunk 11]
```diff
@@ -0,0 +1,270 @@
...
+  display: flex;
+  justify-content: space-between;
+  align-content: center;
+  margin: 20px 0 20px 10px;
+  padding: 20px 0 0 50px;
+}
+section {
+    margin: 20px 0 20px 50px;
+  }
+  p.site-name {
```
Comment (vdaguenet) [inline Javascript-2/Week-1/Movie App/style.css:16]:
```suggestion
p.site-name {
```

The indentation is still funky from here to the end of the file

**Review by vdaguenet — APPROVED** [review]
Thanks for updating the HTML and JS files. It is way better now!
I am approving your PR so you can carry on with the project, but please have a look at the indentation of the CSS file for next week.
